### PR TITLE
Reduce use of Array1S objects

### PR DESCRIPTION
**Pull request by jasondegraw**
Pull request overview
---------------------
Reduces the use of the `Array1S` object, generally replacing the `Array1S` with a reference to an `Array1D` object. This PR mainly removes non-slicing uses of the object, leaving in place 2D to 1D slicing and other similar uses. The changes here should result in no diffs. This is part of the larger effort to migrate to `std::vector`.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Check any new function arguments for performance impacts

### CODE REVIEW

**Comment by mitchute** [comment]
No problems that I see. This looks ready.

File: src/EnergyPlus/OutputProcessor.cc [hunk 1]
```diff
@@ -8068,23 +8123,22 @@ int GetNumMeteredVariables(std::string const &EP_UNUSED(ComponentType), // Given
...
+                         Array1D_string &Names,                                // Variable Names for each variable
+                         int &NumFound                                         // Number Found
 )
 {
 
     // SUBROUTINE INFORMATION:
     //       AUTHOR         Linda Lawrie
     //       DATE WRITTEN   May 2005
-    //       MODIFIED       na
+    //       MODIFIED       Jason DeGraw 2/12/2020, de-optionalized
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputProcessor.cc:8141]: `requirized`?

File: src/EnergyPlus/DataEnvironment.cc [hunk 2]
```diff
@@ -599,7 +599,7 @@ namespace DataEnvironment {
         ShowFatalError("Program terminates due to preceding condition(s).");
     }
 
-    void SetWindSpeedAt(int const NumItems, Array1S<Real64> const Heights, Array1S<Real64> LocalWindSpeed, std::string const &EP_UNUSED(Settings))
+    void SetWindSpeedAt(int const NumItems, const Array1D<Real64> &Heights, Array1D<Real64> &LocalWindSpeed, std::string const &EP_UNUSED(Settings))
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataEnvironment.cc:602]: Why is this an array anyway?  Not your problem, just rambling...

File: src/EnergyPlus/SimulationManager.cc [hunk 3]
```diff
@@ -3047,7 +3047,6 @@ namespace SimulationManager {
                                 EndUses,
                                 Groups,
                                 VarNames,
-                                _,
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimulationManager.cc:3050]: :+1:

**Review by Myoldmopar — APPROVED** [review]
I was expecting to see at least some unit test code change to accommodate this, but I get that it shouldn't really be necessary, so this is fine.  You are touching a lot of code that is unit tested already, so I'd expect to see lots of failures if you broke something.  I have questions about the code you changed, but not the changes you made, just some :man_facepalming: when looking at the original code.  

This is good to go by me.  Develop has now moved a little, so I can pull this branch and merge in develop and run some quick tests locally.

**Comment by Myoldmopar** [comment]
Tested locally, looking good.  Merging.

**Pull request merged**